src/framework/preferences/js/Store.js

```diff
@@ -128,13 +131,14 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         options = options || {};
         var cookieStr = cookieName + "=" + data;
 
-        if (options.expires) {
-            cookieStr += "; expires=" + options.expires;
-        }
-
-        if (options.path) {
-            cookieStr += "; path=" + options.path;
-        }
+        fluid.each(options, function (value, setting) {
```

Comment:
While we are here, could we take the opportunity to improve the docs of fluid.prefs.cookieStore.assembleCookie - e.g. more accurate type for "options" - perhaps Object.<String, String>, or perhaps with a typedef given this argument type appears in more than one function, and a list of at least the settings that we know we support - perhaps with a link to the MDN documentation that you quoted.
Also perhaps fix the minor anomaly that the argument is called "payload" in writeCookie but "data" in assembleCookie. It looks like the argument type in writeCookie must be wrong

Reply:
Other comment issues include the use of "SettingsStore" where this has now been migrated over to DataSources, and the use of "Subcomponent" is anachronistic